lib/rubocop/cop/rspec/aggregate_examples.rb

```diff
@@ -0,0 +1,294 @@
+      #   describe do
+      #     specify do
+      #       expect(number).to be_positive
+      #       expect(number).to be_odd
+      #       is_expected.to be_prime
+      #     end
+      #   end
+      #
+      #   # fair - subject has side effects
+      #   describe do
+      #     specify do
+      #       expect(multiply_by(2)).to be_multiple_of(2)
+      #     end
+      #
+      #     specify do
+      #       expect(multiply_by(3)).to be_multiple_of(3)
+      #     end
+      #   end
+      #
+      #   # The following example will fail if aggregated due to the side
+      #   # effects of the `validate_presence_of` matcher as it leaves an empty
+      #   # comment after itself on the subject making it invalid and the
+      #   # subsequent expectation to fail.
+      #
+      #   # bad, but should not be automatically correctable
+      #   describe do
+      #     it { is_expected.to validate_presence_of(:comment) }
+      #     it { is_expected.to be_valid }
+      #   end
+      #
+      #   # Block expectation syntax is deliberately not supported due to:
+      #   # 1. `subject { -> { ... } }` syntax being hard to detect
+      #   # E.g.:
+      #   it { is_expected.to do_something }
+      #   # looks like an example with non-block syntax, but it might be
+      #   # depending on how the subject is defined. If the subject is defined
+      #   # in a `shared_context`, it's impossible to detect that at all.
+      #   #
+      #   # 2. Aggregation should use composition with an `.and`. Also,
+      #   # aggregation of the `not_to` expectations is barely possible when a
+      #   # matcher doesn't provide a negated variant.
+      #   #
+      #   # 3. Aggregation of block syntax with non-block syntax should be in a
+      #   # specific order.
+      #
+      class AggregateExamples < Cop # rubocop:disable Metrics/ClassLength
+        include RangeHelp
+
+        MSG = 'Aggregate with the example above.'.freeze
+        MSG_FOR_EXPECTATIONS_WITH_SIDE_EFFECTS =
+          "#{MSG} IMPORTANT! Pay attention to the expectation order, some " \
+          'of the matchers have side effects.'.freeze
+
+        MATCHERS_WITH_SIDE_EFFECTS = %w[
+          :validate_presence_of
+          :validate_absence_of
+          :validate_length_of
+          :validate_inclusion_of
+          :validates_exclusion_of
+        ].freeze
+
+        def on_block(node)
+          example_group_with_several_examples(node) do |all_examples|
+            examples = aggregateable_examples(all_examples)
+            next if examples.count < 2
+
+            message = message_for(examples)
+            add_offense(examples[1], location: :expression, message: message)
+          end
+        end
+
+        def autocorrect(example_node)
+          node = example_node.parent
+          examples = aggregateable_examples(node.each_child_node(:block))
+            .select { |child| examples_for_autocorrect?(child) }
+          return if examples.count < 2
```

Comment:
It's a bit weird that there could be an offense to a node that doesn't have examples for auto-correction.
I'm ok with having this guard though.

Reply:
An example of this is:
```
it('be this') { is_expected.to be_this }
it('be that') { is_expected.to be_that }
```
it's not automatically aggregateable, since the docstrings would be lost.